File: 13-06-2024_Padovan_Sequence.cpp
class Solution {
    public:
    int mod = 1e9 + 7;
    int padovanSequence(int n) {
        int curr = 1, next = 1, prev2 = 1, prev = 1;
        for(int i = 3; i <= n; i++){
            next = (prev2 % mod + prev % mod) % mod;
            prev2 = prev % mod;
            prev = curr % mod;
            curr = next % mod;
        }
        return next % mod;
    }
};

File: 13-06-2024_Padovan_Sequence.py
class Solution:
    def padovanSequence(self, n):
        # code here 
        mod = 1000000007
        curr = nex = prev = prev2 = 1
        for i in range(3, n+1):
            nex = (prev2 % mod + prev % mod) % mod
            prev2 = prev % mod
            prev = curr % mod
            curr = nex % mod
        return nex % mod
